### DISCUSSION TITLE
Permission Denied "exec entrypoint.sh"

### DISCUSSION THREAD

**Discussion by ShaunLWM** [community/community#26891] [Actions]
<p>I have recently created an action which utilises <a href="https://play.google.com/store/apps/details?id=com.joaomgcd.join&amp;hl=en_US" rel="nofollow noopener">Join </a>to send notification to devices. The <a href="https://github.com/ShaunLWM/action-join" rel="nofollow noopener">action repository</a> is based off <a href="https://github.com/maddox/actions/tree/master/pushover" rel="nofollow noopener">Pushover</a> which has some values changed.</p>
<p>However, when using my action, I have this error which doesn’t seem to occur in the Pushover action.</p>
<p>Sample run (Pushover only: Pass)</p>
<p><a href="https://github.com/ShaunLWM/awdawd/runs/213770026" rel="nofollow noopener">https://github.com/ShaunLWM/awdawd/runs/213770026</a></p>
<p>Sample run (Join and Pushover - failed):</p>
<p><a href="https://github.com/ShaunLWM/awdawd/runs/213779607" rel="nofollow noopener">https://github.com/ShaunLWM/awdawd/runs/213779607</a></p>
<p>Logs:</p>
<pre><code>Error response from daemon: OCI runtime create failed: container_linux.go:345: starting container process caused "exec: \"/entrypoint.sh\": permission denied": unknown
</code></pre>
<p>I am kind of confused as to why is it happening as it runs off the exact same Dockerfile. </p>
<p>I apologise for being new to this. I’ve tried changing the image in the Dockerfile but to no avail.</p>
<p>Thank you so much.</p>

**Comment by ethomson** [comment, marked as answer]
<p>The runner is trying to run your entrypoint.sh as a script, but it looks like your entrypoint.sh is missing the execute bit.  You’ll need to add it to that file in your action’s repository.</p>
<p>On Linux or macOS, run:</p>
<pre><code>chmod +x entrypoint.sh
git add entrypoint.sh
git commit
</code></pre>
<p>On Windows, run:</p>
<pre><code>git add --chmod=+x -- entrypoint.sh
git commit
</code></pre>
<p>And then push the changes back up.</p>

**Comment by ShaunLWM** [comment]
<p>Ran the command and it’s now working as expected. Thank you so much for your help. I will definitely read up more about the additional paramters for the <code>git add</code> command as this is the first time I’ve seen the argument used. Thank you once again.</p>

**Comment by sefasenturk95** [comment]
<p>This didn’t work for me unfortunately. In GKE i still get an error that says</p>
<pre><code>OCI runtime create failed: container_linux.go:345: starting container
          process caused "exec: \"/server\": permission denied": unknown'
</code></pre>
<p>Also when i try to chmod +x server in the pipeline I get this error:</p>
<pre><code>OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"/bin/sh\": stat /bin/sh: no such file or directory": unknown
123
</code></pre>

**Comment by goodfr** [comment]
<p><a class="mention" href="/sef1995">@sef1995</a> I had the same problem (on Windows) fixed by converting .sh files to lf (instead of crlf).</p>
<p>Best of luck</p>

**Comment by ayushi2705** [comment]
<p>Endings of statements for .sh file and Dockerfile (if you are using) should be LF instead of CRLF. Also  make sure you add #!/bin/bash<br>
in the starting of your .sh file to make it a linux file.</p>
